internal/servers/controller/handlers/accounts/account_service.go

```diff
@@ -312,6 +316,71 @@ func (s Service) createInRepo(ctx context.Context, authMethodId, scopeId string,
 	if out == nil {
 		return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to create user but no error returned from repository.")
 	}
+	return out, nil
+}
+
+func (s Service) createOidcInRepo(ctx context.Context, authMethodId, scopeId string, item *pb.Account) (*oidc.Account, error) {
+	const op = "account_service.(Service).createOidcInRepo"
+	if item == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "nil item")
+	}
+	var opts []oidc.Option
+	if item.GetName() != nil {
+		opts = append(opts, oidc.WithName(item.GetName().GetValue()))
+	}
+	if item.GetDescription() != nil {
+		opts = append(opts, oidc.WithDescription(item.GetDescription().GetValue()))
+	}
+	attrs := &pb.OidcAccountAttributes{}
+	if err := handlers.StructToProto(item.GetAttributes(), attrs); err != nil {
+		return nil, handlers.InvalidArgumentErrorf("Error in provided request.",
+			map[string]string{"attributes": "Attribute fields do not match the expected format."})
+	}
+	a, err := oidc.NewAccount(authMethodId, attrs.GetSubject(), opts...)
+	if err != nil {
+		return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to build user for creation: %v.", err)
+	}
+	repo, err := s.oidcRepoFn()
+	if err != nil {
+		return nil, err
+	}
+
+	out, err := repo.CreateAccount(ctx, scopeId, a)
+	if err != nil {
+		return nil, errors.Wrap(err, op, errors.WithMsg("unable to create user"))
+	}
+	if out == nil {
+		return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to create user but no error returned from repository.")
+	}
+	return out, nil
+}
+
+func (s Service) createInRepo(ctx context.Context, authMethodId, scopeId string, item *pb.Account) (*pb.Account, error) {
+	const op = "account_service.(Service).createInRepo"
```

Comment:
Just an observation that when you call createInRepo you have an AuthMethod and if you passed that, you wouldn't need the scopeId and then you wouldn't have back to back string parameters in the signature.